prototypes/ItemRepeaterPrototypes/ItemsRepeaterExperiments/AttachedBehaviors/SelectionModelPage.xaml.cs

```diff
@@ -0,0 +1,102 @@
+﻿using ItemsRepeaterExperiments.Common;
+using Microsoft.UI.Xaml.Controls;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+
+// The Blank Page item template is documented at https://go.microsoft.com/fwlink/?LinkId=234238
+
+namespace ItemsRepeaterExperiments.AttachedBehaviors
+{
+    /// <summary>
+    /// An empty page that can be used on its own or navigated to within a Frame.
+    /// </summary>
+    public sealed partial class SelectionModelPage : Page
+    {
+        private List<Team> teams = DataSourceCreator<Team>.CreateRandomizedList(100);
+        private SelectionModel selectionModel = new SelectionModel();
+        public SelectionModelPage()
+        {
+            this.InitializeComponent();
+            selectionModel.Source = teams;
+            selectionModel.SelectionChanged += SelectionModel_SelectionChanged;
+            selectionModel.SingleSelect = true;
+        }
+
+
+        /// <summary>
+        /// Attach properties to virtualized items and update containers
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="args"></param>
+        private void ItemsRepeater_ElementPrepared(Microsoft.UI.Xaml.Controls.ItemsRepeater sender, Microsoft.UI.Xaml.Controls.ItemsRepeaterElementPreparedEventArgs args)
+        {
+            new PointerBehaviors(args.Element as Control).Click += Item_Click;
+            var selectionbehavior = new SelectionBehavior(args.Element as Control);
+
+            if(selectionModel.SelectedIndex != null && selectionModel.SelectedIndex.GetAt(0) == args.Index)
+            {
+                selectionbehavior.IsSelected = true;
+            }
+            else
+            {
+                selectionbehavior.IsSelected = false;
+            }
+        }
+
+        /// <summary>
+        /// This event gets raised when an item gets selected
+        /// @WINUI This event definitely needs a list of selected and deselected items
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="args"></param>
+        private void SelectionModel_SelectionChanged(SelectionModel sender, SelectionModelSelectionChangedEventArgs args)
+        {
+            // Having the deselected items would really be handy here ...
+            foreach (var item in teams)
```

Comment:
Yeah.. Iterating over all the data is not feasible. This list can be huge. If we had the api for just getting the realized containers, we could use that to re-evaluate all the realized containers. If args had information on what changed, we could use that to get containers that have changed and update just those.

Reply:
One optimization one could do is just cut of the for loop if we have reached realized containers and then again encounter null, but that's not a good way to handle this either.